CONTRIBUTING.md

```diff
@@ -0,0 +1,158 @@
+# Contributing
```

Comment:
TODO: add text and flowchart diagram indicating where changes should be located based on #214